lib/auth/auth.go

```diff
@@ -1350,6 +1359,99 @@ func (a *AuthServer) DeleteRole(name string) error {
 	return a.Access.DeleteRole(name)
 }
 
+func (a *AuthServer) CreateAccessRequest(req services.AccessRequest) error {
+	if err := a.validateAccessRequest(req); err != nil {
+		return trace.Wrap(err)
+	}
+	ttl, err := a.calculateMaxAccessTTL(req)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	now := a.clock.Now().UTC()
+	req.SetCreationTime(now)
+	exp := now.Add(ttl)
+	// Set acccess expiry if an allowable default was not provided.
+	if req.GetAccessExpiry().Before(now) || req.GetAccessExpiry().After(exp) {
+		req.SetAccessExpiry(exp)
+	}
+	// By default, resource expiry should match access expiry.
+	req.SetExpiry(req.GetAccessExpiry())
+	// If the access-request is in a pending state, then the expiry of the underlying resource
+	// is capped to to PendingAccessDuration in order to limit orphaned access requests.
+	if req.GetState().IsPending() {
+		pexp := now.Add(defaults.PendingAccessDuration)
+		if pexp.Before(req.Expiry()) {
+			req.SetExpiry(pexp)
+		}
+	}
+	return a.DynamicAccess.CreateAccessRequest(req)
+}
+
+// validateAccessRequest checks if a given access request is allowable based
```

Comment:
We should probably move this logic to `lib/services/access_request.go`. This will mirror how the RBAC logic sits in `lib/services/role.go`.